languages/go/src/main/kotlin/io/vrap/codegen/languages/go/model/GoFileProducer.kt

```diff
@@ -0,0 +1,382 @@
+        |        return err
+        |    }
+        |    $fieldHandlers
+        |    return nil
+        |}
+        """.trimMargin()
+    }
+
+    fun ObjectType.renderDiscriminatorMapper(): String {
+
+        val caseStatements = allAnyTypes.getTypeInheritance(this)
+            .filterIsInstance<ObjectType>()
+            .filter { !it.discriminatorValue.isNullOrEmpty() }
+            .map {
+                val extraAttrs = it.renderUnmarshalFuncFields("nil")
+                """
+            |   case "${it.discriminatorValue}":
+            |       obj := ${it.name}{}
+            |       if err := decodeStruct(input, &obj); err != nil {
+            |           return nil, err
+            |       }
+            |       <$extraAttrs>
+            |       return obj, nil
+            """.trimMargin()
+            }
+            .joinToString("\n")
+
+        return """
+            |func mapDiscriminator$name(input interface{}) ($name, error) {
+            |
+            |    var discriminator string
+            |    if data, ok := input.(map[string]interface{}); ok {
+            |        discriminator, ok = data["${this.discriminator()}"].(string)
+            |        if !ok {
+            |            return nil, errors.New("Error processing discriminator field '${this.discriminator()}'")
+            |        }
+            |    } else {
+            |        return nil, errors.New("Invalid data")
+            |    }
+            |
+            |    switch discriminator {
+            |    $caseStatements
+            |    }
+            |    return nil, nil
+            |}
+        """.trimMargin()
+    }
+
+    // Renders the attribute of this model as type annotations.
+    fun ObjectType.renderStructFields(): String {
+        return goStructFields(true)
+            .map {
+                val comment: String = it.type.toLineComment().escapeAll()
+                var name = it.name
+
+                // Make sure we don't have an Error field and Method
+                if (name.lowercase() == "error" && this.isErrorObject()) {
+                    name = "${name}Message"
+                }
+
+                if (it.required) {
+                    """
+                    |<$comment>
+                    |${name.exportName()} ${it.type.renderTypeExpr()} `json:"${it.name}"`
+                    """.trimMargin()
+                } else {
+                    val type = it.type
+
+                    val pointer = when (type) {
+                        is ArrayType -> ""
+                        is ObjectType -> if (type.isDiscriminated()) "" else "*"
+                        is AnyType -> if (type.renderTypeExpr() == "interface{}") "" else "*"
+                        else -> "*"
+                    }
+
+                    """
+                    |<$comment>
+                    |${name.exportName()} ${pointer}${type.renderTypeExpr()} `json:"${it.name},omitempty"`
```

Comment:
I'm facing a special case with the Commercetools State Transitions field. It behaves differently when the field is set, not set or empty. For this case, `omitempty` should be remove for the Transitions field.

Or maybe there is another way to deal with it?

See issue https://github.com/labd/commercetools-go-sdk/issues/72

Reply:
Should be fixed with https://github.com/commercetools/rmf-codegen/pull/219/commits/b162a010628b438c3186c97c9fbd0f36204fd4a2